### DISCUSSION TITLE
After upgrade can't delete device

### DISCUSSION THREAD

**Discussion by ahmedovelshan** [netbox-community/netbox#9197] [Q&A]
After upgrade I can't delete device. Error below

<class 'django.db.utils.IntegrityError'>

update or delete on table "dcim_device" violates foreign key constraint "netbox_onboarding_on_device_id_cee3ef7b_fk_dcim_devi" on table "netbox_onboarding_onboardingdevice"
DETAIL:  Key (id)=(248) is still referenced from table "netbox_onboarding_onboardingdevice".

Netbox Version v3.1.6

Could someone help me to solve this problem?

**Comment by candlerb** [comment]
Do you have some sort of plugin installed? Called "netbox unboarding" or similar?

**Reply by ahmedovelshan** [reply]
Yes we have used this in old version. But it was disabled in new config

Plugin name: NetBox Onboarding plugin

**Reply by candlerb** [reply]
Looks like you'll have to delete the plugin's tables then; I would guess it has added foreign key relations to the `dcim_device` table that don't allow for ON DELETE CASCADE.
